src/hu_stuff.c

```diff
@@ -1232,12 +1242,22 @@ static void HU_widget_build_coord (void)
 static void HU_widget_build_fps (void)
 {
   char hud_fpsstr[HU_MAXLINELENGTH/4];
-  extern int fps;
 
   sprintf(hud_fpsstr,"\x1b%c%d \x1b%cFPS", '0'+CR_GRAY, fps, '0'+CR_ORIG);
   HUlib_add_string_to_cur_line(&w_fps, hud_fpsstr);
 }
 
+static void HU_widget_build_rate (void)
+{
+  char hud_ratestr[HU_MAXLINELENGTH];
+
+  sprintf(hud_ratestr, "Sprites %4d Segs %4d Visplanes %4d FPS %3d",
+          rendered_vissprites, rendered_segs, rendered_visplanes, fps);
+  HUlib_add_string_to_cur_line(&w_rate, hud_ratestr);
+  sprintf(hud_ratestr, " Voxels %4d", rendered_voxels);
```

Comment:
This will show "0 voxels" in 99.99%. Could we only add this line if voxels were successfully initialized?

Reply:
It shows something 100% of the time for me these days 😄 I thought this cheat was meant for developers. But of course, if you don't like it, I'll keep it for myself.

Reply:
Sure it's useful for developers. I didn't mean to say I don't like it or find it useless per-se. It's just useless to print "0 Voxels" on a second line if not a single voxel was loaded at start up. 😉

Reply:
Fixed in https://github.com/fabiangreffrath/woof/pull/1254/commits/0f9eb3d8276e7b128e6c518eb1aacacfabc4ca1c

Reply:
Perfect, thanks!

Now, I'd also like to use this as a guard around `VX_NearbySprites()` (and probably `VX_ProjectVoxel()`) to save us from some recursions and lookups if no Voxels are available.

Reply:
Done in https://github.com/fabiangreffrath/woof/pull/1254/commits/632769c6d4ae3db6aa34089cfa3f095276f3d8da `VX_ProjectVoxel()` is already guarded.

By the way, I'm still not sure why `VX_NearbySprites()` is needed. Maybe it's a vanilla problem?

Reply:
In my understanding, this is to add Voxels of nearby objects which only reach into your field of view, because Voxels expand into all three dimensions -- in contrast to sprites, which are lacking depth.

Reply:
> In my understanding, this is to add Voxels of nearby objects which only reach into your field of view, because Voxels expand into all three dimensions -- in contrast to sprites, which are lacking depth.

But this effect is hard to notice in practice.

Obviously, a large number of voxel models on the screen reduces performance, so I experimented with various ways to reduce their number. Apart from dynamic reducing `VX_MAX_DIST`, nothing looks good.

Reply:
If you IDCLIP through monsters, you can see parts of their Voxels in front of your view, although you have already passed through the monster's center, so it should be behind your view plane by BSP logic. Maybe that's the effect, I'm not sure.